src/v/kafka/server/usage_manager.cc

```diff
@@ -208,33 +228,56 @@ ss::future<> usage_manager::accounting_fiber::start() {
                               return close_window();
                           }).finally([this] {
             if (!_gate.is_closed()) {
-                _timer.arm(
-                  ss::lowres_clock::now() + _usage_window_width_interval);
+                _timer.arm(_usage_window_width_interval);
             }
         });
     });
-    const auto now = ss::lowres_clock::now();
+
+    /// Calculate the next time the timer should fire, this depends on the most
+    /// recent buckets open timestamp.
+    vassert(_buckets[_current_window].is_open(), "Current window is closed");
+    const auto most_recent_window_start = ss::lowres_system_clock::time_point(
+      std::chrono::seconds(_buckets[_current_window].begin));
+    const auto delta = ss::lowres_system_clock::now()
+                       - most_recent_window_start;
     vassert(
-      last_window_delta <= _usage_window_width_interval,
+      delta > 0s && delta <= _usage_window_width_interval,
       "Error correctly detecting last window delta");
-    _timer.arm((now + _usage_window_width_interval) - last_window_delta);
-    _persist_disk_timer.arm(now + _usage_disk_persistance_interval);
+    _timer.arm(_usage_window_width_interval - delta);
+    _persist_disk_timer.arm(_usage_disk_persistance_interval);
 }
 
-std::vector<usage_window>
-usage_manager::accounting_fiber::get_usage_stats() const {
+ss::future<std::vector<usage_window>>
+usage_manager::accounting_fiber::get_usage_stats(bool include_open) {
```

Comment:
what does `include_open` means. include_open_window?

Reply:
yeah, grab the stats of now from the most recent window close

Reply:
changing to `include_open_window`